internal/multigitter/review.go

```diff
@@ -0,0 +1,253 @@
+package multigitter
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/lindell/multi-gitter/internal/scm"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+// Reviewer reviews pull requests in an organization
+type Reviewer struct {
+	VersionController VersionController
+	FeatureBranch     string
+	Comment           string
+	All               bool
+	Batch             string
+	Pager             string
+	DisablePaging     bool
+	IncludeApproved   bool
+}
+
+const (
+	header        = "# *********************************************************"
+	reviewOptions = "[a/r/c/n/d/h (help)]"
+	batchApprove  = "approve"
+	batchReject   = "reject"
+	batchComment  = "comment"
+)
+
+type approvdPr struct {
+	scm.PullRequest
+	approved bool
+}
+
+// Reviewer reviews pull requests in an organization
+func (s Reviewer) Review(ctx context.Context) error {
+	prs, err := s.VersionController.GetPullRequests(ctx, s.FeatureBranch)
+	if err != nil {
+		return err
+	}
+
+	var approvedPrs []approvdPr
+
+	for _, pr := range prs {
+		approved, err := s.VersionController.IsPullRequestApprovedByMe(ctx, pr)
+		if err != nil {
+			log.Errorf("Failed to retrieve pull request reviews: %s", err.Error())
+			continue
+		}
+
+		if approved && !s.IncludeApproved {
+			continue
+		}
+
+		approvedPrs = append(approvedPrs, approvdPr{
+			PullRequest: pr,
+			approved:    approved,
+		})
+	}
+
+	if len(approvedPrs) == 0 {
+		fmt.Println("All pull requests are approved by you. You can still view and comment them by using review --include-approved")
+		return nil
+	}
+
+	var reviewDiffs string
+
+	for _, pr := range approvedPrs {
+		diff, err := s.VersionController.DiffPullRequest(ctx, pr.PullRequest)
+		if err != nil {
+			log.Errorf("Error occurred while retrieving diff: %s", err.Error())
+			continue
+		}
+
+		if !s.All && s.Batch == "" {
+			if err := s.leaveReview(ctx, strings.NewReader(fmt.Sprintf("%s\n# %s\n%s\n%s", header, pr.String(), header, diff)), pr); err != nil {
+				log.Errorf("Error occurred while reviewing pr: %s", err.Error())
+			}
+		} else {
+			reviewDiffs = fmt.Sprintf("%s\n%s\n# %s\n%s\n%s", reviewDiffs, header, pr.String(), header, diff)
+		}
+	}
+
+	if s.All || s.Batch != "" {
+		return s.leaveReviews(ctx, strings.NewReader(reviewDiffs), approvedPrs...)
+	}
```

Comment:
A suggested improvement. The assumption is that All and Batch will be mutually inclusive. 
Removes some duplication of the Sprintf, and will avoid slow additions to reviewDiffs for large diffs.

```suggestion
	var reviewDiffs strings.Builder

	for _, pr := range approvedPrs {
		diff, err := s.VersionController.DiffPullRequest(ctx, pr.PullRequest)
		if err != nil {
			log.Errorf("Error occurred while retrieving diff: %s", err.Error())
			continue
		}

		reviewDiff := fmt.Sprintf("%s\n# %s\n%s\n%s", header, pr.String(), header, diff)

		if !s.All {
			if err := s.leaveReview(ctx, reviewDiff, pr); err != nil {
				log.Errorf("Error occurred while reviewing pr: %s", err.Error())
			}
		} else {
			reviewDiffs.WriteString("\n")
			reviewDiffs.WriteString(reviewDiff)
		}
	}

	if s.All {
		return s.leaveReviews(ctx, reviewDiffs.String(), approvedPrs...)
	}
```